webpack.config.dev.js

```diff
@@ -39,39 +40,34 @@ module.exports = {
     fs: 'empty'
   },
   module: {
-    loaders: [
+    rules: [
       {
-        exclude: /(node_modules|public)/,
-        loader: 'babel',
-        query: {
-          'plugins': [
-            ['react-transform', {
-              'transforms': [{
-                'transform': 'react-transform-hmr',
-                'imports': ['react'],
-                'locals': ['module'],
-              }],
-            }],
-          ],
-        },
         test: /\.jsx?$/,
-      },
-      {
-        loader: 'json-loader',
-        test: /\.json$/,
+        exclude: /(node_modules|public)/,
+        use: [{
+          loader: 'babel-loader',
+          options: {
+            plugins: [
+              ['react-transform', {
+                'transforms': [{
+                  'transform': 'react-transform-hmr',
+                  'imports': ['react'],
+                  'locals': ['module'],
+                }]
+              }]
+            ]
+          }
+        }]
       }
-    ],
+    ]
   },
   resolve: {
-    extensions: ['', '.js', '.jsx'],
+    extensions: ['.js', '.jsx'],
   },
-  progress: true,
   plugins: [
     // hot reload
     new webpack.HotModuleReplacementPlugin(),
 
-    new webpack.NoErrorsPlugin(),
```

Comment:
> Add [bail](https://webpack.js.org/api/cli/#advanced-options) option? (ref: [noEmitOnErrors](https://webpack.js.org/configuration/optimization/#optimization-noemitonerrors))

Reply:
In which situation would that help? I'm a little reluctant to add options unless at least someone understands what it does, and I frankly don't.
I put big syntax errors in the code and adding `--bail` did not make the `npm run build` fail...

Reply:
GTM. 
> I thought that it could be used as replacement for webpack.NoErrorsPlugin() but this doesn't seem to be the solution.